src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Processors/Settings/GeneralizeSetting.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+using Hl7.FhirPath;
+using Microsoft.Health.Fhir.Anonymizer.Core.AnonymizerConfigurations;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Anonymizer.Core.Processors.Settings
+{
+    public class GeneralizeSetting
+    {
+        public JObject Cases { get; set; }
+        public GeneralizationOtherValuesOperation OtherValues { get; set; }
+
+        public static GeneralizeSetting CreateFromRuleSettings(Dictionary<string, object> ruleSettings)
+        {
+            EnsureArg.IsNotNull(ruleSettings);
+
+            JObject cases = null;
+            GeneralizationOtherValuesOperation otherValues = GeneralizationOtherValuesOperation.redact;
+            if (ruleSettings.ContainsKey(RuleKeys.Cases))
+            {
+                try
+                {
+                    cases = JObject.Parse(ruleSettings.GetValueOrDefault(RuleKeys.Cases)?.ToString());
+                }
+                catch (Exception ex)
+                {
+                    throw new AnonymizerConfigurationErrorsException($"Invalid cases {RuleKeys.Cases}", ex);
+                }  
+            }
+
+            if (ruleSettings.ContainsKey(RuleKeys.OtherValues))
+            {
+                Enum.TryParse(ruleSettings.GetValueOrDefault(RuleKeys.OtherValues)?.ToString(), true, out otherValues);
```

Comment:
The default othervalues might get overwritten if parsing failed. Better to assign the default value when parse failed (in the else block).

Reply:
The exception will be thrown if parsing failed when validate setting?